packages/mds-transactions-service/tests/service.spec.ts

```diff
@@ -0,0 +1,217 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import { TransactionServiceManager } from '../service/manager'
+import { TransactionServiceClient } from '../client'
+import { TransactionRepository } from '../repository'
+
+describe('Transaction Repository Tests', () => {
+  beforeAll(async () => {
+    await TransactionRepository.initialize()
+  })
+
+  it('Run Migrations', async () => {
+    await TransactionRepository.runAllMigrations()
+  })
+
+  it('Revert Migrations', async () => {
+    await TransactionRepository.revertAllMigrations()
+  })
+
+  afterAll(async () => {
+    await TransactionRepository.shutdown()
+  })
+})
+
+const TransactionServer = TransactionServiceManager.controller()
+
+const device_id = 'ee6bf5c7-bce0-46c9-a5c9-8652724059d7'
+const provider_id = '3452fa87-bfd7-42c5-9c53-5e07bde13671'
+const unknown_provider_id = '654b106d-5706-495f-8c89-64e17a5a3ed8'
+const transaction_id = '37bd96ac-69bd-4634-9b22-ff081d7a5a09'
+const unknown_transaction_id = '822415eb-baaa-40ff-b219-a5ed214e2114'
+const receipt_id = 'a5eb612e-a154-4339-a760-aee95908dc51'
+const operation_id = '4fcbbd4f-c0cb-46b7-b7dd-55ebae535493'
+const status_id = '15c99c65-cf78-46a9-9055-c9973e43f061'
+const receipt = { receipt_id, timestamp: Date.now(), receipt_details: {}, origin_url: '' }
+
+const malformed_uuid = '176b8453-ccaf-41c7-a4df-f7b3f80bddd1xxxxxxx'
+
+describe('Transaction Service Tests', () => {
+  beforeAll(async () => {
+    await TransactionServer.start()
+  })
+
+  it('Post Good Transaction', async () => {
+    const transaction = await TransactionServiceClient.createTransaction({
+      transaction_id,
+      provider_id,
+      device_id,
+      timestamp: Date.now(),
+      amount: 100, // "I'd buy THAT for a dollar!"
+      fee_type: 'base_fee',
+      receipt
+    })
+    expect(transaction.device_id).toEqual(device_id)
+    expect(transaction.transaction_id).toEqual(transaction_id)
+  })
+
+  it('Post Transaction with malformed transaction_id', async () => {
+    try {
+      await TransactionServiceClient.createTransaction({
+        transaction_id: malformed_uuid,
+        provider_id,
+        device_id,
+        timestamp: Date.now(),
+        amount: 100, // "I'd buy THAT for a dollar!"
+        fee_type: 'base_fee',
+        receipt
+      })
+      expect('did not happen').toBe('happened')
+    } catch (err) {
+      expect(err.type).toBe('ValidationError')
```

Comment:
Feel like I'm beating a dead horse, but jest _specifically_ has fixtures for testing promise rejection, as has been mentioned previously

Reply:
It was a step backwards in clarity for me